### PR TITLE
F8 supporting driver changes [2/5]

### PR DESCRIPTION
**Pull request by xinlipn**
sl/f8_driver is branched off sl/f8 branch. It's meant for F8 driver related changes

### CODE REVIEW

File: include/miopen/miopen.h [hunk 1]
```diff
@@ -339,7 +345,9 @@ typedef enum
         4, /*!< Pack of four 8-bit int points in NCHW_VECT_C format (Partially supported) */
     miopenBFloat16 = 5, /*!< 16-bit binary floating point (8-bit exponent, 7-bit fraction)
                            (Partially supported) */
-    miopenDouble = 6,   /*!< 64-bit floating point (Partially supported) */
+    miopenDouble  = 6,  /*!< 64-bit floating point (Partially supported) */
+    miopenFloat8  = 7,
+    miopenBFloat8 = 8
```
Comment (atamazov) [inline include/miopen/miopen.h:350]: [Recommendation] Please add documentation - explain the difference between FP8 and BF8.

File: include/miopen/miopen.h [hunk 2]
```diff
@@ -573,6 +581,8 @@ typedef enum
     MIOPEN_CONVOLUTION_ATTRIB_DETERMINISTIC =
         1, /*!< Restrict MIOpen convolutions to kernels which produce numerically deterministic
               results. 0 - disabled (default), 1 - enabled >*/
+    MIOPEN_CONVOLUTION_ATTRIB_FP8_ROUNDING_MODE = 2,
+    MIOPEN_CONVOLUTION_ATTRIB_COMPUTE_TYPE      = 3,
```
Comment (atamazov) [inline include/miopen/miopen.h:585]: Please add doc - explain what is "compute type".

File: include/miopen/miopen.h [hunk 3]
```diff
@@ -111,6 +111,12 @@ typedef enum
     miopenStatusVersionMismatch = 10, /*!< Version mismatch of the supplied binary data argment. */
 } miopenStatus_t;
 
+typedef enum
+{
+    miopenF8RoundingModeStandard   = 0,
+    miopenF8RoundingModeStochastic = 1,
```
Comment (atamazov) [inline include/miopen/miopen.h:117]: [Recommendation] Please add doc - explain what is standard mode and what is stochastic one.

**Review by atamazov — CHANGES_REQUESTED** [review]
Partial review. The PR requires the https://github.com/ROCmSoftwarePlatform/MIOpen/labels/IMPACTS_API label.

**Comment by xinlipn** [comment]
Current PR has dependency on this PR
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2134

File: include/miopen/miopen.h (hunk 1 above)
Comment (xinlipn) [reply include/miopen/miopen.h:350]: To put in an one liner. Float8(FP8) or BFloat8(BF8) are two 8-bit floating point data types. 
* FP8: 1-4-3 (sign-exponent-mantissa), more precision, smaller range
* BF8: 1-5-2 (sign-exponent-mantissa), less precision, larger range

File: include/miopen/miopen.h (hunk 2 above)
Comment (xinlipn) [reply include/miopen/miopen.h:585]: > Please add doc - explain what is "compute type".

GEMM operation is defined as D = alpha * A * B + beta * C, where A*B is dot product.

Comment (JehandadKhan) [reply include/miopen/miopen.h:585]: @atamazov  It represents the accumulator type for accumulation in convolution.

File: include/miopen/miopen.h (hunk 3 above)
Comment (JehandadKhan) [reply include/miopen/miopen.h:117]: This represents how numbers are rounded from a higher precision type to FP8

Comment (atamazov) [reply include/miopen/miopen.h:117]: @JehandadKhan Thanks! I am guessing the same but what I am thinking about is adding this to the doxygen doc. Does it sound reasonable?

File: include/miopen/miopen.h (hunk 2 above)
Comment (atamazov) [reply include/miopen/miopen.h:585]: @JehandadKhan @xinlipn We already have this feature in some other convolutions (e.g. FP16) and in some other primitives (e.g. FP16 pooling). For so far it is hidden, i.e. not exposed in the public API. And I am not sure if it is worth to make it public for them.

If we are going to use this attribute for F8 only, then I recommend either documenting this or (better) renaming the attribute.

File: include/miopen/miopen.h (hunk 1 above)
Comment (atamazov) [reply include/miopen/miopen.h:350]: @xinlipn Thanks! What I am thinking about is adding this to the doxygen doc. Does it sound reasonable for you?

File: include/miopen/miopen.h [hunk 4]
```diff
@@ -573,6 +581,8 @@ typedef enum
     MIOPEN_CONVOLUTION_ATTRIB_DETERMINISTIC =
         1, /*!< Restrict MIOpen convolutions to kernels which produce numerically deterministic
               results. 0 - disabled (default), 1 - enabled >*/
+    MIOPEN_CONVOLUTION_ATTRIB_FP8_ROUNDING_MODE = 2,
```
Comment (atamazov) [inline include/miopen/miopen.h:584]:
```suggestion
    MIOPEN_CONVOLUTION_ATTRIB_F8_ROUNDING_MODE = 2,
```
because this is applicable to both FP8 and BF8.

**Pull request closed without merging**